In [1]:
import pandas as pd

In [5]:
df_sales = pd.read_csv('../dataset/vendas_2023_2024.csv')
df_sales.head()

,id,id_client,id_product,qtd,total,sale_date
0,0,42,105,11,3405.0,2023-09-10
1,1,3,136,9,16873.9,15-09-2024
2,2,25,139,7,9475.3,2024-08-13
3,4,20,23,5,55893.0,2023-02-03
4,5,8,57,4,451403.9,2024-02-12


In [10]:
df_sales.dtypes

id              int64
id_client       int64
id_product      int64
qtd             int64
total         float64
sale_date         str
dtype: object

In [8]:
rows_count, cols_count = df_sales.shape

print(f'Total de linhas: {rows_count}')
print(f'Total de colunas: {cols_count}')

Total de linhas: 9895
Total de colunas: 6


## 1- Visão Geral do Dataset

O dataset possui 9895 registros de vendas com 6 colunas (id, id_client, id_product, qtde, total e sale_date).
A partir do título do arquivo e com uma exploração visual podemos dizer que as vendas aconteceram num período entre 2023 e 2024. Mas não conseguimos dizer o intervalo exato, pois seria necessário padronizar os formatos das datas. Algumas seguem yyyy-MM-dd, outras dd-MM-yyyy.

In [16]:
df_sales['total'].agg(['min', 'max', 'mean']).map('{:.2f}'.format)

min         294.50
max     2222973.00
mean     263797.83
Name: total, dtype: str

# 2 - Análise de Valores Númericos 